frame/support/src/storage/generator/double_map.rs

```diff
@@ -431,33 +457,111 @@ impl<
 	}
 }
 
+/// Test iterators for StorageDoubleMap
 #[cfg(test)]
-mod test {
-	use sp_io::TestExternalities;
-	use crate::storage::{self, StorageDoubleMap};
-	use crate::hash::Twox128;
+#[allow(dead_code)]
+mod test_iterators {
+	use codec::{Encode, Decode};
+	use crate::storage::{generator::StorageDoubleMap, IterableStorageDoubleMap, unhashed};
+
+	pub trait Trait {
+		type Origin;
+		type BlockNumber;
+	}
+
+	crate::decl_module! {
+		pub struct Module<T: Trait> for enum Call where origin: T::Origin {}
+	}
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode)]
+	struct NoDef(u32);
+
+	crate::decl_storage! {
+		trait Store for Module<T: Trait> as Test {
+			DoubleMap: double_map hasher(blake2_128_concat) u16, hasher(blake2_128_concat) u32 => u64;
+		}
+	}
+
+	fn key_before_prefix(mut prefix: Vec<u8>) -> Vec<u8> {
+		let last = prefix.iter_mut().last().unwrap();
+		assert!(*last != 0, "mock function not implemented for this prefix");
```

Comment:
You are aware that rust panics automatically with `debug_assertions` enabled? (we have this in CI)

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0b5b6e7c75f5163808b6fb16a6a2c1ae

Reply:
And if you wanted to keep it more explicit, `*last = last.checked_sub(1).unwrap()` would be the way to go.

But using `unwrap_or_else(|| unimplemented!())` is really weird.

Reply:
But we can keep this now.

Reply:
> You are aware that rust panics automatically with `debug_assertions` enabled? (we have this in CI)
> 
> https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=0b5b6e7c75f5163808b6fb16a6a2c1ae

I though CI was running in release mode, and so operation was just silently wrapping around.

But I guess best was: `*last = last.checked_sub(1).expect("not implementation for mock")` indeed

Reply:
Yes in release, but with `debug-assertions` enabled: https://github.com/paritytech/substrate/blob/master/.gitlab-ci.yml#L191